homeassistant/components/tedee/__init__.py

```diff
@@ -0,0 +1,80 @@
+"""Init the tedee component."""
+import logging
+
+from pytedee_async import TedeeAuthException, TedeeClientException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.network import NoURLAvailableError, get_url
+
+from .const import CONF_HOME_ASSISTANT_ACCESS_TOKEN, DOMAIN
+from .coordinator import TedeeApiCoordinator
+from .views import TedeeWebhookView
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+PLATFORMS = [
+    Platform.LOCK,
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Integration setup."""
+    entry.async_on_unload(entry.add_update_listener(options_update_listener))
+
+    hass.data.setdefault(DOMAIN, {})[
+        entry.entry_id
+    ] = coordinator = TedeeApiCoordinator(hass, entry)
+
+    await coordinator.async_config_entry_first_refresh()
+
+    home_assistant_token = entry.data.get(CONF_HOME_ASSISTANT_ACCESS_TOKEN, "")
+    # Setup webhook if long lived access token
+    if home_assistant_token != "":
```

Comment:
```suggestion
    home_assistant_token = entry.data.get(CONF_HOME_ASSISTANT_ACCESS_TOKEN)
    # Setup webhook if long lived access token
    if home_assistant_token is not None:
```

Personally I would prefer this

Reply:
Also, what actually would happen when no token is found?

Reply:
I think I test for "" because it might be the empty string coming from the options flow. If no token is found, then the websocket part is not initialized and we only rely on polls for updates.